components/App.js

```diff
@@ -0,0 +1,84 @@
+App = React.createClass({
+
+	getInitialState() {
+		return {
+			loading: false,
+			searchingText: '',
+			gif: {}
+		};
+	},
+
+	handleSearch: function(searchingText) {
+		var self = this;
+		self.setState({
+			loading: true  // 2.
+		});
+		self.getGif(searchingText, function(gif) {  // 3.
+			self.setState({
+				loading: false,
+				gif: gif,
+				searchingText: ""  // c
+			});
+//    }).bind(this);
+		});
+	},
+
+	getGif: function(searchingText, callback) {  // 1.
+		var GIPHY_API_URL = "http://api.giphy.com";
```

Comment:
Przerzuc to jako const na gore pliku lub zmien konwencje nazewnictwa - z wielkich liter tylko stale globalne